kernel/memory/src/lib.rs

```diff
@@ -99,6 +107,37 @@ pub fn create_contiguous_mapping<F: Into<PteFlagsArch>>(
 }
 
 
+/// A convenience function that creates a new memory mapping by allocating specific frames from physical memory.
+/// Returns the new `MappedPages`.
+/// 
+/// # Locking / Deadlock
+/// Currently, this function acquires the lock on the frame allocator and the kernel's `MemoryManagementInfo` instance.
+/// Thus, the caller should ensure that the locks on those two variables are not held when invoking this function.
+pub fn map_frame_range<F: Into<PteFlagsArch>>(
+    range: FrameRange,
+    flags: F,
+) -> Result<MappedPages, &'static str> {
+    let kernel_mmi_ref = get_kernel_mmi_ref().ok_or("map_range(): KERNEL_MMI was not yet initialized!")?;
+    let num_frames = range.size_in_frames();
+    let allocated_pages = allocate_pages(num_frames).ok_or("memory::map_range(): couldn't allocate contiguous pages!")?;
+    let allocated_frames = allocate_frames_by_bytes_at(range.start_address(), num_frames)
```

Comment:
this is incorrect, you're passing in the number of frames to a function that expects the number of bytes.
```suggestion
    let allocated_frames = allocate_frames_at(range.start_address(), num_frames)
```

Reply:
Good catch, thank you